nextcord/interactions.py

```diff
@@ -1252,3 +1257,44 @@ class InteractionMessage(_InteractionMessageMixin, Message):
     """
 
     pass
+
+
+class MessageInteraction:
+    """Represents a message's interaction data, regardless of application.
+
+    A message's interaction data is a property of a message when the message
+    is a response to an interaction from any bot.
+
+    .. versionadded:: 2.0
+
+    Attributes
+    -----------
+    data: :class:`dict`
+        The raw data from the interaction.
+    id: :class:`int`
+        The interaction's ID.
+    type: :class:`InteractionType`
+        The interaction type.
+    name: :class:`str`
+        The application ID that the interaction was for.
+    user: :class:`User`
+        The user or that sent the interaction;
+    """
+
+    __slots__ = (
+        "_state",
+        "data",
+        "id",
+        "type",
+        "name",
+        "user",
+    )
+
+    def __init__(self, *, data: MessageInteractionPayload, state: ConnectionState):
+        self._state: ConnectionState = state
+
+        self.data: MessageInteractionPayload = data
+        self.id: int = int(data["id"])
+        self.type: int = data["type"]
+        self.name: str = data["name"]
+        self.user: User = User(data=data["user"], state=state)
```

Comment:
Here's one way to get the member (tested).
You can put back all the good changes you reverted.
```suggestion
        if "member" in data and guild is not None:
            member_user = {**data["member"], "user": data["user"]}
            self.user = Member(state=self._state, guild=guild, data=member_user)  # type: ignore
        else:
            self.user = User(state=self._state, data=data["user"])
```

Reply:
Gotcha